src/Core/Url/QueryParam.php

```diff
@@ -72,11 +72,15 @@ public function isRaw()
 
     /**
      * Generate the paramatere to be appended to the url
-     * @return string the parameter on this format: ``name=value``
+     * @return string the parameter on this format: ``name=value`` or ``name``
      */
     public function generate()
     {
-        return $this->getName() . '=' . $this->getValue();
+        $result = $this->getName();
+        if (!is_null($this->value)) {
+            $result .= '=' . $this->getValue();
+        }
+        return $result;
```

Comment:
For ease of reviewing you should place this change in its own PR (or at least it's own commit in the PR), mixing things when reviewing is hard for maintainers and bad for history. The commit is about how resolve url works, and this is about how url parameters are rendered